### PR TITLE
Log events

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

File: vmhost/hostCore/execution.go [hunk 1]
```diff
@@ -332,6 +333,14 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 			isChildComplete = true
 			return
 		}
+		switch input.CallType {
```
Comment (sasurobert) [inline vmhost/hostCore/execution.go:336]: make a small separate function for this.

File: vmhost/hostCore/host.go [hunk 2]
```diff
@@ -577,3 +587,12 @@ func (host *vmHost) logFromGasTracer(functionName string) {
 		}
 	}
 }
+
+func (host *vmHost) CompleteLogEntriesWithCallType(vmOutput *vmcommon.VMOutput, callType string) {
+	for _, logEntry := range vmOutput.Logs {
+		_, containsId := host.transferLogIdentifiers[string(logEntry.Identifier)]
+		if containsId {
+			logEntry.Data[0] = []byte(callType)
```
Comment (sasurobert) [inline vmhost/hostCore/host.go:595]: I think we can even make a break here.

File: vmhost/hostCore/host.go [hunk 3]
```diff
@@ -577,3 +587,12 @@ func (host *vmHost) logFromGasTracer(functionName string) {
 		}
 	}
 }
+
+func (host *vmHost) CompleteLogEntriesWithCallType(vmOutput *vmcommon.VMOutput, callType string) {
```
Comment (sasurobert) [inline vmhost/hostCore/host.go:591]: add comment on public function

File: vmhost/hostCore/execution.go [hunk 4]
```diff
@@ -332,6 +333,14 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
...
+		case vm.AsynchronousCall:
+			host.CompleteLogEntriesWithCallType(vmOutput, "AsyncCall")
+		case vm.AsynchronousCallBack:
+			host.CompleteLogEntriesWithCallType(vmOutput, "AyncCallback")
+		default:
+			host.CompleteLogEntriesWithCallType(vmOutput, "ExecuteOnDestContext")
+		}
 	}
 
 	isChildComplete = true
```
Comment (sasurobert) [inline vmhost/hostCore/execution.go:346]: do not create transferValueOnly log event if you already created one on lines 336-343

File: vmhost/hostCore/execution.go [hunk 5]
```diff
@@ -348,6 +351,17 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 	return
 }
 
+func (host *vmHost) CompleteLogEntriesAfterBuiltinCall(input *vmcommon.ContractCallInput, vmOutput *vmcommon.VMOutput) {
```
Comment (sasurobert) [inline vmhost/hostCore/execution.go:354]: add comment on public function.

**Review by andrei-marinica — DISMISSED** [review]

**Review by CostinCarabas — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by CostinCarabas — APPROVED** [review]

**Pull request merged**